picard/log.py

```diff
@@ -177,11 +179,15 @@ def name_filter(record):
             path = path.resolve().relative_to(picard_module_path)
         except ValueError:
             pass
-        else:
-            if not DebugOpt.PLUGIN_FULLPATH.enabled and 'plugins' in path.parts:
-                parts = list(reversed(path.parts))
-                parts = parts[:parts.index('plugins') + 1]
-                path = Path(*reversed(parts))
+
+    if path.is_absolute() and not DebugOpt.PLUGIN_FULLPATH.enabled:
+        try:
+            path = path.resolve().relative_to(USER_PLUGIN_DIR)
+            parts = list(p for p in path.parts if not p.endswith('.zip'))
+            parts.insert(0, 'plugins')
```

Comment:
I don't understand this insert.


```python
>>> path = Path('/a/b/c/subdir/e/f.zip/g')
>>> USER_PLUGIN_DIR = Path('/a/b/c/subdir')
>>> path = path.resolve().relative_to(USER_PLUGIN_DIR)
>>> str(path)
'e/f.zip/g'
>>> parts = list(p for p in path.parts if not p.endswith('.zip'))
>>> parts
['e', 'g'] 
>>> parts.insert(0, 'plugins')
>>> parts
['plugins', 'e', 'g']
>>> path = Path(*parts)
>>> str(path)
'plugins/e/g'
```

Why would we introduce an element which isn't in any of existing paths?

Reply:
> Why would we introduce an element which isn't in any of existing paths?

It sort of is in the original `USER_PLUGIN_DIR` in Picard, but gets stripped out when the path is resolved relative to `USER_PLUGIN_DIR`.  My thinking was to add it back in to clearly indicate in the log that this was in a plugin.

I'll remove it.